modules/operations/cycle_highway_tag.js

```diff
@@ -21,7 +22,15 @@ export function operationCycleHighwayTag(context, selectedIDs) {
     'highway/track',
     'highway/unclassified',
     'highway/tertiary',
-    'line'
+    'line',
+    'crossing/unmarked;crossing:markings=no',
+    'crossing/marked;crossing:markings=yes',
+    'crossing/marked;crossing:markings=zebra',
+    'crossing/marked;crossing:markings=lines',
+    'crossing/marked;crossing:markings=ladder',
+    'crossing/marked;crossing:markings=dashes',
+    'crossing/marked;crossing:markings=dots',
+    'crossing/marked;crossing:markings=ladder:skewed',
   ];
 
   // same selection as before?
```

Comment:
Later on in this file, try to step through the code at line 33 and examine how you might refactor it.  It's currently just testing one case- highways- using the existing `allowPresetRegex` and `defaultPresetIDs` arrays. 

To make use of the new `allowHighwayPresetRegex` and `allowCrossingPresetRegex` you will need to test each condition separately: 

* Is the preset a match for allowCrossingPresetRegex? If so, cycle its tags using the crossing preset array for crossings. 
* Else, is the preset a match for allowHighwayPresetRegex? If so, then instead cycle its tags using the crossing preset array for highways. 
* Otherwise, do nothing.

Reply:
> Later on in this file, try to step through the code at line 33 and examine how you might refactor it. It's currently just testing one case- highways- using the existing `allowPresetRegex` and `defaultPresetIDs` arrays.
> 
> To make use of the new `allowHighwayPresetRegex` and `allowCrossingPresetRegex` you will need to test each condition separately:
> 
> * Is the preset a match for allowCrossingPresetRegex? If so, cycle its tags using the crossing preset array for crossings.
> * Else, is the preset a match for allowHighwayPresetRegex? If so, then instead cycle its tags using the crossing preset array for highways.
> * Otherwise, do nothing.

Thanks for providing me with more context to the implementation. I will introduce the conditionals.

I am not clear on this "Later on in this file, try to step through the code at line 33 and examine how you might refactor it."